src/components/Button/Button.vue

```diff
@@ -0,0 +1,381 @@
+<!--
+ - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @author Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @license GNU AGPL version 3 or any later version
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<docs>
+
+### General description
+General purpose button component. See props for different options.
+[Use material design icons only for icons](https://www.npmjs.com/package/vue-material-design-icons) and remember to set their size to 20.
+
+### Usage
+```
+<Button
+	text="Example text"
+	color="primary"
+	aria-label="Example aria label" />
+```
+
+</docs>
+
+<template>
+	<button
+		v-ripple
```

Comment:
If this directive mandatory? I find it annoying to be honest.

Reply:
Yes, for now all the buttons will have it, this has been approved by the design team. I think it provides a nice click visual feedback. What is it that annoys you about it? :)

Reply:
Well, maybe "annoying" is the wrong wording. To me it looks to playful and not professional and kind of distracts me with to much movement (especially that the background color expansion starts at the click position). That's the subjective part of my critics 😉 

As for the objective part: I haven't seen this feedback anywhere yet, so the visual metaphor of providing feedback is quite uncommon, I think. I furthermore suspect that it will work badly when using the keyboard only, because there is no click event when pressing enter. So, I guess the ripple-directive will do nothing and you end up with different feedback for the same action.

It would be nice if the netlify docs would work properly though, this would help test it.

Reply:
Addon: The last release of this directive as well as the last commit to its repo is four years old (2017) and we would get a dependency to vue@2.5.8 (and no vue@3 support for sure). I would consider this unmaintained, and look for an alternative. @skjnldsv @nextcloud/vuejs

Reply:
So maybe at least use this directive: https://github.com/justintaddei/v-wave (last release two months old, supports Vue3, seems to do the same as `v-ripple`)

Reply:
Oh that could be a blocker yes. Thanks for the suggestion, I will look into this directive.
As per the keyboard navigation, I added a fallback behavior, there's a feedback but it's not as cool, it's just a "pressed" state with the change of colors.

I will try to update the docs soon. Do you know if there's any way to include MD icons in the docs??

Reply:
> I will try to update the docs soon. Do you know if there's any way to include MD icons in the docs??

Yes, it's quite straight-forward. We have it for the Actions component for example:
https://github.com/nextcloud/nextcloud-vue/blob/593b3bb411f4c9be8acf9eab8249f5ec351034ef/src/components/Actions/Actions.vue#L84-L123

Reply:
Thanks @raimund-schluessler :)